roles/mariadb/tasks/main.yml

```diff
@@ -1,24 +1,41 @@
 ---
-- block:
-  - name: Add MariaDB APT key
-    apt_key:
-      keyserver: "{{ mariadb_keyserver }}"
-      id: "{{ mariadb_keyserver_id }}"
+- name: Download MariaDB GPG key
+  ansible.builtin.get_url:
+    url: "https://mariadb.org/mariadb_release_signing_key.asc"
+    dest: "/tmp/mariadb_release_signing_key.asc"
+    mode: '0644'
+
+- name: Add the MariaDB GPG key to apt
+  ansible.builtin.shell:
+    cmd: "apt-key add /tmp/mariadb_release_signing_key.asc"
+  args:
+    removes: "/tmp/mariadb_release_signing_key.asc"
```

Comment:
Can we use the built-in url feature of `apt_key`? https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_key_module.html

```yaml
- name: Add an Apt signing key, uses whichever key is at the URL
  ansible.builtin.apt_key:
    url: https://ftp-master.debian.org/keys/archive-key-6.0.asc
    state: present
```

Reply:
```
fatal: [default]: FAILED! => {"after": ["D94AA3F0EFE21092", "871920D1991BC93C"], "before": ["D94AA3F0EFE21092", "871920D1991BC93C"], "changed": true, "fp": "AED4B06F473041FA", "id": "AED4B06F473041FA", "key_id": "AED4B06F473041FA", "msg": "apt-key did not return an error, but failed to add the key (check that the id is correct and *not* a subkey)", "short_id": "473041FA"}
```

Reply:
Try https://mariadb.com/kb/en/installing-mariadb-deb-files-with-ansible/#importing-mariadb-gpg-key


```yaml
- name: Add an apt key by id from a keyserver
  ansible.builtin.apt_key:
    keyserver: hkp://keyserver.ubuntu.com:80
    id: 0xF1656F24C74CD1D8
```